install: Add network header override

kylemanna opened the pull request:
- If a network header file is found, use that for generating the
  network file.
- This functionality allows users to create a header file @
  `/etc/systemd/network/template/dosync-${interface}.network.head` to
  override the default.
- Example use case: Adding `Tunnel=name-of-tunnel` to the eth0
  interface file to signal systemd-networkd to bring up a tunnel using
  the eth0 interface.
- Resolves #16

gh2o commented:
I think it would be a better idea to place the extra content at the end of the generated content instead. It seems slightly awkward to have to rewrite the file header and the Match section.

kylemanna committed:
install: Add ability to append network config

* If a network tail file is found, append to the generated config.
* This functionality allows users to create a tail file @
  /etc/systemd/network/template/dosync-${interface}.network.tail to
  add additional options to the network config.
* Example use case: Adding "Tunnel=name-of-tunnel" to the eth0
  interface file to signal systemd-networkd to bring up a tunnel using
  the eth0 interface.

kylemanna commented:
That makes sense.  I've updated branch to do just that.  Code is simpler now.

gh2o commented:
Merged. Thanks for your suggestion!

kylemanna commented:
Thank you for starting such an excellent project!
